python/ray/experimental/channel/nccl_group.py

```diff
@@ -262,6 +262,9 @@ def allreduce(
         if self._closed:
             raise RayChannelError("NCCL group has been destroyed.")
 
+        assert (
```

Comment:
Add a test for this case?

Reply:
Do you mean adding a unit test for `allreduce`? We don't have a way to make the assertion fail in e2e tests because the `recv_buf` is [automatically derived](https://github.com/ray-project/ray/blob/5483b0e0f07030062c11bb86a711c2539f1892ad/python/ray/dag/collective_node.py#L126) from `send_buf`.

Reply:
update the error message to state that this is a RayCG issue.

Reply:
updated: https://github.com/ray-project/ray/pull/49300/commits/11193a6cbf3dad404662180a4ec7fe08a73253a3